File: Assets/Scripts/Menu.js
﻿#pragma strict

var grappleCursor : Texture2D;
var menuCursor : Texture2D;

private var map : Camera;

function Start() {
	Cursor.SetCursor(grappleCursor, new Vector2(16, 16), CursorMode.Auto);
}

function Update () {
	if (Input.GetKeyDown(KeyCode.Escape)) {
		if (Time.timeScale == 1) {Time.timeScale = 0;} else {Time.timeScale = 1;}
	}
}

function Respawn() {Application.LoadLevel(Application.loadedLevel);}

File: Assets/Scripts/Generic/Camera/Menu.cs
using UnityEngine;
using UnityEngine.SceneManagement;

public class Menu : MonoBehaviour {
	public Texture2D grappleCursor;
	public Texture2D menuCursor;

	float normalTimeScale = 1;

	void Start() {
		Cursor.SetCursor(grappleCursor, Vector2.one * 16, CursorMode.Auto);
	}

	void Update() {
		if (Input.GetKeyDown(KeyCode.Escape)) {
			if (Time.timeScale == 0) Time.timeScale = normalTimeScale;
			else Time.timeScale = 0;
		}
	}

	///Reload current scene
	public void Restart() {
		int thisIndex = SceneManager.GetActiveScene().buildIndex;
		SceneManager.LoadScene(thisIndex);
	}

	///deprecated
	public void Respawn() {Restart();}
}